inlong-agent/agent-plugins/src/test/java/org/apache/inlong/agent/plugin/sources/TestBinlogReader.java

```diff
@@ -18,43 +18,70 @@
+
 public class TestBinlogReader {
 
     private static Gson gson = new Gson();
 
     @Test
     public void testDebeziumFormat() {
         String debeziumJson = "{\n"
-            + "    \"before\": null,\n"
-            + "    \"after\": {\n"
-            + "      \"id\": 1004,\n"
-            + "      \"first_name\": \"Anne\",\n"
-            + "      \"last_name\": \"Kretchmar\",\n"
-            + "      \"email\": \"annek@noanswer.org\"\n"
-            + "    },\n"
-            + "    \"source\": {\n"
-            + "      \"version\": \"1.8.1.Final\",\n"
-            + "      \"name\": \"dbserver1\",\n"
-            + "      \"server_id\": 0,\n"
-            + "      \"ts_sec\": 0,\n"
-            + "      \"gtid\": null,\n"
-            + "      \"file\": \"mysql-bin.000003\",\n"
-            + "      \"pos\": 154,\n"
-            + "      \"row\": 0,\n"
-            + "      \"snapshot\": true,\n"
-            + "      \"thread\": null,\n"
-            + "      \"db\": \"inventory\",\n"
-            + "      \"table\": \"customers\"\n"
-            + "    },\n"
-            + "    \"op\": \"r\",\n"
-            + "    \"ts_ms\": 1486500577691\n"
-            + "  }";
+                + "    \"before\": null,\n"
+                + "    \"after\": {\n"
+                + "      \"id\": 1004,\n"
+                + "      \"first_name\": \"Anne\",\n"
+                + "      \"last_name\": \"Kretchmar\",\n"
+                + "      \"email\": \"annek@noanswer.org\"\n"
+                + "    },\n"
+                + "    \"source\": {\n"
+                + "      \"version\": \"1.8.1.Final\",\n"
+                + "      \"name\": \"dbserver1\",\n"
+                + "      \"server_id\": 0,\n"
+                + "      \"ts_sec\": 0,\n"
+                + "      \"gtid\": null,\n"
+                + "      \"file\": \"mysql-bin.000003\",\n"
+                + "      \"pos\": 154,\n"
+                + "      \"row\": 0,\n"
+                + "      \"snapshot\": true,\n"
+                + "      \"thread\": null,\n"
+                + "      \"db\": \"inventory\",\n"
+                + "      \"table\": \"customers\"\n"
+                + "    },\n"
+                + "    \"op\": \"r\",\n"
+                + "    \"ts_ms\": 1486500577691\n"
+                + "  }";
         DebeziumFormat debeziumFormat = gson
-            .fromJson(debeziumJson, DebeziumFormat.class);
+                .fromJson(debeziumJson, DebeziumFormat.class);
         Assert.assertEquals("customers", debeziumFormat.getSource().getTable());
     }
+
+    // @Test
+    public void binlogStartSpacialTest() throws Exception {
+        JobProfile jobProfile = new JobProfile();
+        jobProfile.set(BinlogReader.JOB_DATABASE_USER, "root");
+        jobProfile.set(BinlogReader.JOB_DATABASE_PASSWORD, "123456");
+        jobProfile.set(BinlogReader.JOB_DATABASE_HOSTNAME, "9.135.12.10");
```

Comment:
remove the IP, it's better to mock the UT.

Reply:
resolve